### PR TITLE
UIOAIPMH-71: OAI-PMH settings: Ensure your settings HTML page title follow this format - <<App name>> settings - <<selected page name>> - FOLIO

### PR DESCRIPTION
**Pull request by UladzislauKutarkin**
[UIOAIPMH-71](https://issues.folio.org/browse/UIOAIPMH-71)
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by vashjs — APPROVED** [review]

**Review by Dmytro-Melnyshyn — APPROVED** [review]

File: src/settings/index.js [hunk 1]
```diff
@@ -1,8 +1,11 @@
 import React from 'react';
-import { FormattedMessage } from 'react-intl';
+import { FormattedMessage, injectIntl } from 'react-intl';
 
 import { Settings } from '@folio/stripes/smart-components';
 
+import { TitleManager } from '@folio/stripes-core';
+import { stripesShape } from '@folio/stripes/core';
```
Comment (zburke) [inline src/settings/index.js:7]:
```
import { TitleManager, stripesShape } from '@folio/stripes/core`;
```
Note that's `stripes/core` so you're importing via the `@folio/stripes` wrapper package.

File: src/settings/index.js [hunk 2]
```diff
@@ -36,14 +48,41 @@ export default class OaiPmhSettings extends React.Component {
...
+        case path.includes('sets'):
+          return intl.formatMessage({ id: 'ui-oai-pmh.settings.sets.manager' });
+        case path.includes('technical'):
+          return intl.formatMessage({ id: 'ui-oai-pmh.settings.technical.manager' });
+        case path.includes('logs'):
+          return intl.formatMessage({ id: 'ui-oai-pmh.settings.logs.manager' });
+        default:
+          return intl.formatMessage({ id: 'ui-oai-pmh.settings.title.manager' });
+      }
+    };
```
Comment (zburke) [inline src/settings/index.js:71]: A little easier to maintain and less copy/paste-y as that list grows:
```
const findLabelByRoute = (path) => {
  const list = ['behavior', 'general', 'sets', 'technical', 'logs'];
  return `ui-oai-pmh.settings.${list.find(i => i === path) ?? 'title'}.manager`;
};
```

**Review by zburke — CHANGES_REQUESTED** [review]
Please fix the import. Consider the simpler `fineLabelByRoute` function. Everything else looks great. Thanks for adding tests!

File: src/settings/index.js (hunk 2 above)
Comment (UladzislauKutarkin) [reply src/settings/index.js:71]: Thanks for that, fixed

File: src/settings/index.js (hunk 1 above)
Comment (UladzislauKutarkin) [reply src/settings/index.js:7]: Done!

**Pull request merged**